### PR TITLE
(CODEMGMT-62) Automate Tests for "Basic Workflow" User Scenarios

### PR DESCRIPTION
**Pull request by cowofevil**
Add more basic workflow user scenario integration tests. Changed the pre-suites to be ordered and added a new pre-suite for installing data generation tool. Updated test run scripts to utilize the new ordered pre-suites.

### CODE REVIEW

File: integration/tests/user_scenario/basic_workflow/single_env_unicode_paths.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#Tests
+step 'Deploy "production" Environment via r10k'
+on(master, 'r10k deploy environment -v')
+
+#Note: Usually a full Puppet Run would be performed for verification.
+#Since Puppet has problems with Unicode, this test will verify the file
+#directly in the r10k environment.
+
+agents.each do |agent|
```
Comment (zreichert) [inline integration/tests/user_scenario/basic_workflow/single_env_unicode_paths.rb]: Do Better Ryan.
Should be on master, not on agents.

Comment (elyscape) [inline integration/tests/user_scenario/basic_workflow/single_env_unicode_paths.rb]: This seems unnecessarily aggressive.  The unedited version more so.

Comment (cowofevil) [inline integration/tests/user_scenario/basic_workflow/single_env_unicode_paths.rb]: @elyscape Inside joke, we sit next to each other.

Comment (zreichert) [inline integration/tests/user_scenario/basic_workflow/single_env_unicode_paths.rb]: @elyscape You are correct, @cowofevil sorry for my inappropriate comment.

**Comment by cowofevil** [comment]
@zreichert So the Unicode test passes for me when running on the vm pooler. Merge and pray?

**Comment by zreichert** [comment]
Tests did not work locally.  
ENV variables LANG and LC_CTYPE 
Merging ... cross your fingers

**Pull request merged**